### PR TITLE
Validation Changes for Java Artifacts

### PR DESCRIPTION
**Pull request by harrryr**
*Issue #, if available:*
New Java Artifacts will be released. The new release includes significant changes in the metric schema of the telemetry. We need to update the validation templates to reflect the new expected changes. This branch is also currently being used by the cwagent repo to perform release testing.

May 30, 2024 Update:
Java Artifacts have been released. 
- Updated file name from appsignals -> application-signals
- Removed code forcing old version of artifact

Test run: https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/9309496601
Test run after review: https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/9309836295
3rd Test run: https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/9324083316

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+      fail-fast: false
+      matrix:
+        aws-region: ['us-east-1']
+    uses: ./.github/workflows/appsignals-e2e-eks-test.yml
+    secrets: inherit
+    with:
+      aws-region: ${{ matrix.aws-region }}
+      test-cluster-name: 'e2e-playground'
+      caller-workflow-name: "eks-test"
+      appsignals-adot-image-name: '417921506511.dkr.ecr.us-east-1.amazonaws.com/aws-opentelemetry-agent/metric-schema-changes'
```
Comment (majanjua-amzn) [inline .github/workflows/test.yml:138]: Your test.yml is not using the same image as is being used by the release workflow:
`611364707713.dkr.ecr.us-west-2.amazonaws.com/adot-autoinstrumentation-java-operator-staging:1.33.0-SNAPSHOT-91cbba8`

Let's rerun this with the correct value for extra confidence.

File: validator/src/main/resources/expected-data-template/eks/aws-sdk-call-metric.mustache [hunk 2]
```diff
@@ -123,51 +108,20 @@
...
-      name: RemoteTarget
-      value: ::s3:::e2e-test-bucket-name-{{testingId}}
-
--
-  metricName: Latency
-  namespace: {{metricNamespace}}
-  dimensions:
+      name: RemoteResourceType
+      value: AWS::S3::Bucket
     -
```
Comment (majanjua-amzn) [inline validator/src/main/resources/expected-data-template/eks/aws-sdk-call-metric.mustache:122]: Incorrectly removed the following metric aggregation: `[Environment, Service, RemoteService, RemoteResourceIdentifier, RemoteResourceType]`

This happened in both EKS and EC2 for Java

File: .github/workflows/application-signals-java-e2e-eks-test.yml [hunk 3]
```diff
@@ -30,14 +30,15 @@ permissions:
...
-  # The precense of this env var is required for use by terraform and AWS CLI commands
+  # The presence of this env var is required for use by terraform and AWS CLI commands
   # It is not redundant
   AWS_DEFAULT_REGION: ${{ inputs.aws-region }}
   ENABLEMENT_SCRIPT_S3_BUCKET: ${{ secrets.APP_SIGNALS_E2E_ENABLEMENT_SCRIPT }}
   SAMPLE_APP_FRONTEND_SERVICE_IMAGE: ${{ secrets.APP_SIGNALS_E2E_TEST_ACC }}.dkr.ecr.${{ inputs.aws-region }}.amazonawss.com/${{ secrets.APP_SIGNALS_E2E_FE_SA_IMG }}
   SAMPLE_APP_REMOTE_SERVICE_IMAGE: ${{ secrets.APP_SIGNALS_E2E_TEST_ACC }}.dkr.ecr.${{ inputs.aws-region }}.amazonawss.com/${{ secrets.APP_SIGNALS_E2E_RE_SA_IMG }}
-  METRIC_NAMESPACE: AppSignals
-  LOG_GROUP_NAME: /aws/appsignals/eks
+  CW_AGENT_IMAGE_RELEASE_VERSION: public.ecr.aws/y8s3a7r9/cloudwatch-agent:latest
```
[the remark this answers is not in the document]
Comment (harrryr) [reply .github/workflows/application-signals-java-e2e-eks-test.yml:39]: Removed

File: .github/workflows/application-signals-java-e2e-eks-test.yml [hunk 4]
```diff
@@ -173,8 +168,20 @@ jobs:
...
+              # kubectl patch deploy -namazon-cloudwatch amazon-cloudwatch-observability-controller-manager --type='json' \
+              # -p='[{"op": "replace", "path": "/spec/template/spec/containers/0/args/1", "value": "--auto-instrumentation-java-image=${{ inputs.appsignals-adot-image-name }}"}]'
+              # echo "LOG: Describing operator deployment for the second time"
+              # kubectl patch amazoncloudwatchagents -n amazon-cloudwatch cloudwatch-agent --type='json' -p='[{"op": "replace", "path": "/spec/image", "value": "${{ env.CW_AGENT_IMAGE_RELEASE_VERSION }}"}]'
+              
+              execute_and_retry 2 "kubectl delete pods --all -n amazon-cloudwatch"
+              execute_and_retry 2 "kubectl wait --for=condition=Ready pod --all -n amazon-cloudwatch"
+          
+              execute_and_retry 2 "kubectl delete pods --all -n ${{ env.SAMPLE_APP_NAMESPACE }}"
+              execute_and_retry 2 "kubectl wait --for=condition=Ready --request-timeout '5m' pod --all -n ${{ env.SAMPLE_APP_NAMESPACE }}"
```
[the remark this answers is not in the document]
Comment (harrryr) [reply .github/workflows/application-signals-java-e2e-eks-test.yml:184]: Removed

File: .github/workflows/application-signals-java-e2e-eks-test.yml [hunk 5]
```diff
@@ -173,8 +168,20 @@ jobs:
...
               60
           
-              execute_and_retry 2 "kubectl delete pods --all -n ${{ env.SAMPLE_APP_NAMESPACE }}" "" 60
-              execute_and_retry 2 "kubectl wait --for=condition=Ready --request-timeout '5m' pod --all -n ${{ env.SAMPLE_APP_NAMESPACE }}" "" 10
+              # Patch ADOT and CW Agent images, then restart the CW pods (Will leave it here for now for future releases)
+              # echo "LOG: Describing operator deployment"
+              # kubectl describe deploy -namazon-cloudwatch amazon-cloudwatch-observability-controller-manager
+              # echo "LOG: Patching operator deployment with ADOT image"
+              # kubectl patch deploy -namazon-cloudwatch amazon-cloudwatch-observability-controller-manager --type='json' \
+              # -p='[{"op": "replace", "path": "/spec/template/spec/containers/0/args/1", "value": "--auto-instrumentation-java-image=${{ inputs.appsignals-adot-image-name }}"}]'
```
[the remark this answers is not in the document]
Comment (harrryr) [reply .github/workflows/application-signals-java-e2e-eks-test.yml:176]: Removed

File: validator/src/main/java/com/amazon/aoc/App.java [hunk 6]
```diff
@@ -135,9 +135,19 @@ public class App implements Callable<Integer> {
...
 
+  @CommandLine.Option(
+          names = {"--instance-id"},
+          defaultValue = "defaultId")
+  private String instanceId;
+
+  @CommandLine.Option(
+          names = {"--private-dns-name"},
+          defaultValue = "defaultDnsName")
+  private String privateDnsName;
```
[the remark this answers is not in the document]
Comment (harrryr) [reply validator/src/main/java/com/amazon/aoc/App.java:149]: Removed

File: .github/workflows/application-signals-java-e2e-eks-test.yml [hunk 7]
```diff
@@ -172,9 +166,9 @@ jobs:
               ${{ env.SAMPLE_APP_NAMESPACE }} && \
               aws eks update-kubeconfig --name ${{ inputs.test-cluster-name }} --region ${{ inputs.aws-region }}" \
               60
-          
-              execute_and_retry 2 "kubectl delete pods --all -n ${{ env.SAMPLE_APP_NAMESPACE }}" "" 60
-              execute_and_retry 2 "kubectl wait --for=condition=Ready --request-timeout '5m' pod --all -n ${{ env.SAMPLE_APP_NAMESPACE }}" "" 10
+                  
+              execute_and_retry 2 "kubectl delete pods --all -n ${{ env.SAMPLE_APP_NAMESPACE }}"
+              execute_and_retry 2 "kubectl wait --for=condition=Ready --request-timeout '5m' pod --all -n ${{ env.SAMPLE_APP_NAMESPACE }}"
```
Comment (majanjua-amzn) [inline .github/workflows/application-signals-java-e2e-eks-test.yml:171]: Why?

**Review by majanjua-amzn — APPROVED** [review]

**Pull request merged**